source/glkernel/include/glkernel/sample.hpp

```diff
@@ -207,6 +207,33 @@ size_t poisson_square(tkernel<glm::tvec2<T, P>> & kernel, const T min_dist, cons
     return k + 1;
 }
 
+template <typename T, glm::precision P>
+size_t stratified(tkernel<glm::tvec2<T, P>> & kernel)
+{
+	assert(kernel.depth() == 1);
+
+	auto stratum_width = 1.0 / kernel.width();
+	auto stratum_height = 1.0 / kernel.height();
+
+	std::random_device RD;
+	std::mt19937_64 generator(RD());
+	std::uniform_real_distribution<> x_dist(0.0, stratum_width);
+	std::uniform_real_distribution<> y_dist(0.0, stratum_height);
+
+	size_t k = 0;
+	for (auto x = 0; x < kernel.width(); ++x)
+	{
+		for (auto y = 0; y < kernel.height(); ++y)
+		{
+			auto x_coord = x * stratum_width + x_dist(generator);
+			auto y_coord = y * stratum_height + y_dist(generator);
+			auto sample = glm::tvec2<T, P>(x_coord, y_coord);
+			kernel[k++] = sample;
+		}
+	}
+	return k;
```

Comment:
this function probably does not need a return value, since it always returns the kernel.size() (at least for the 2D case it is defined with depth = 1).

so either return kernel.size() or preferably remove return value and adjust function signature